File: BoxOfDice/GameMode.swift
//
//  GameMode.swift
//  BoxOfDice
//

import Foundation

enum GameMode: String, CaseIterable, Identifiable {
    case classic
    case speedRun
    case bigBox
    case bigBoxSpeed
    case passAndPlay

    var id: String { rawValue }

    var title: String {
        switch self {
        case .classic:     return L10n.string("Classic")
        case .speedRun:    return L10n.string("Speed Run")
        case .bigBox:      return L10n.string("Big Box")
        case .bigBoxSpeed: return L10n.string("Big Box Speed")
        case .passAndPlay: return L10n.string("Pass & Play")
        }
    }

    var subtitle: String {
        switch self {
        case .classic:     return L10n.string("12 tiles · 2 dice · lowest score wins")
        case .speedRun:    return L10n.string("12 tiles · 2 dice · score + elapsed seconds")
        case .bigBox:      return L10n.string("18 tiles · 3 dice · lowest score wins")
        case .bigBoxSpeed: return L10n.string("18 tiles · 3 dice · score + elapsed seconds")
        case .passAndPlay: return L10n.string("2–4 players · each plays one Classic round")
        }
    }

    var tileCount: Int {
        switch self {
        case .classic, .speedRun, .passAndPlay: return 12
        case .bigBox, .bigBoxSpeed:             return 18
        }
    }

    var baseDiceCount: Int {
        switch self {
        case .classic, .speedRun, .passAndPlay: return 2
        case .bigBox, .bigBoxSpeed:             return 3
        }
    }

    var isTimed: Bool {
        self == .speedRun || self == .bigBoxSpeed
    }

    var isMultiplayer: Bool {
        self == .passAndPlay
    }

    var group: Group {
        switch self {
        case .classic, .speedRun:   return .classic
        case .bigBox, .bigBoxSpeed: return .bigBox
        case .passAndPlay:          return .multiplayer
        }
    }

    func finalScore(baseScore: Int, elapsedSeconds: Int) -> Int {
        baseScore + (isTimed ? elapsedSeconds : 0)
    }

    enum Group: String {
        case classic     = "Classic"
        case bigBox      = "Big Box"
        case multiplayer = "Multiplayer"
    }
}

File: app/src/main/java/com/example/boxofdice/model/GameMode.kt
package com.example.boxofdice.model

enum class GameMode(
    val displayName: String,
    val description: String,
    val tileCount: Int,
    val diceCount: Int,
    val hasTimer: Boolean,
    val isMultiplayer: Boolean = false
) {
    CLASSIC(
        displayName = "Classic",
        description = "12 tiles · 2 dice",
        tileCount = 12,
        diceCount = 2,
        hasTimer = false
    ),
    SPEED_RUN(
        displayName = "Speed Run",
        description = "12 tiles · 2 dice · timed",
        tileCount = 12,
        diceCount = 2,
        hasTimer = true
    ),
    BIG_BOX(
        displayName = "Big Box",
        description = "18 tiles · 3 dice",
        tileCount = 18,
        diceCount = 3,
        hasTimer = false
    ),
    BIG_BOX_SPEED(
        displayName = "Big Box Speed",
        description = "18 tiles · 3 dice · timed",
        tileCount = 18,
        diceCount = 3,
        hasTimer = true
    ),
    PASS_AND_PLAY(
        displayName = "Pass & Play",
        description = "2–4 players · one Classic round each",
        tileCount = 12,
        diceCount = 2,
        hasTimer = false,
        isMultiplayer = true
    )
}
